python/ray/util/client/dataclient.py

```diff
@@ -97,9 +97,8 @@ def _data_main(self) -> None:
                 # here.
                 logger.info("Server disconnected from data channel")
             else:
-                logger.error(
-                    f"Got Error from data channel -- shutting down: {e}")
-                raise e
+                logger.exception(
+                    "Got Error from data channel -- shutting down:")
```

Comment:
Shouldn't we still raise the exception?

Reply:
Nope! We are already exiting the thread!

Reply:
If we both log and raise, than the error is duplicated (once in the form of the log and another time in something like `Error in Thread-7....<full stack trace again>`